ote/src/cljs/ote/views/own_services.cljs

```diff
@@ -66,22 +70,46 @@
                     (tr [:buttons :delete])])]}
       (tr [:dialog :delete-transport-service :confirm] {:name name})])])
 
-(defn transport-services-table-rows [e! services transport-operator-id]
-  [ui/table-body (merge {:class "table-body"}
-                        {:display-row-checkbox false})
+(defn- service-errors
+  [{::t-service/keys [transport-type interface-types sub-type name] :as service}]
+  (when (and (= sub-type :schedule) (not ((set interface-types) :route-and-schedule)))
+    (let [tr-types (set transport-type)]
+      (cond
+        (tr-types :road)
+        {:name name
+         :error :no-schedule-road}
+        (tr-types :sea)
+        {:name name
+         :error :no-schedule-sea}))))
+
+(defn transport-services-table-rows [e! services transport-operator-id info-open?]
+  [ui/table-body {:class "table-body"
+                  :display-row-checkbox false}
    (doall
-     (map-indexed
-       (fn [i {::t-service/keys [id type published name]
-               ::modification/keys [created modified] :as row}]
-         ^{:key i}
+     (map
+       (fn [{::t-service/keys [id type sub-type interface-types published name]
+             ::modification/keys [created modified] :as row}]
+         ^{:key id}
          [ui/table-row {:selectable false :display-border false}
           [ui/table-row-column
            [:a (merge {:href (str "/#/edit-service/" id)
                        :on-click #(do
                                     (.preventDefault %)
                                     (e! (fp/->ChangePage :edit-service {:id id})))}
                       (stylefy/use-sub-style style-base/front-page-service-table :link)) name]]
-          [ui/table-row-column {:class "hidden-xs "} (tr [:field-labels :transport-service ::t-service/published?-values (not (nil? published))])]
+          [ui/table-row-column {:class "hidden-xs "}
+           (let [errors (service-errors row)]
```

Comment:
maybe if-let?

Reply:
just if was enough the whole errors symbol wasn't used anywhere. 👍